tests/api/v2/handlers/test_operations_api.py

```diff
@@ -0,0 +1,437 @@
+                          'jitter': '2/8',
+                          'visibility': 50,
+                          'auto_close': False,
+                          'obfuscator': 'plain-text',
+                          'use_learning_parsers': False}
+
+    return expected_operation
+
+
+@pytest.fixture
+def test_agent(loop):
+    agent = Agent(paw='123', sleep_min=2, sleep_max=8, watchdog=0, executors=['sh'], platform='linux')
+    loop.run_until_complete(BaseService.get_service('data_svc').store(agent))
+    return agent
+
+
+@pytest.fixture
+def test_executor(test_agent):
+    return ExecutorSchema().load(dict(timeout=60, platform=test_agent.platform, name='linux',
+                                      command='d2hvYW1p'))
+
+
+@pytest.fixture
+def test_ability(test_executor, loop):
+    ability = AbilitySchema().load(dict(ability_id='123',
+                                        tactic='discovery',
+                                        technique_id='auto-generated',
+                                        technique_name='auto-generated',
+                                        name='Manual Command',
+                                        description='test ability',
+                                        executors=[ExecutorSchema().dump(test_executor)]))
+    loop.run_until_complete(BaseService.get_service('data_svc').store(ability))
+    return ability
+
+
+@pytest.fixture
+def active_link(test_executor, test_agent, test_ability):
+    return {
+        'command': test_executor.command,
+        'paw': test_agent.paw,
+        'ability': test_ability,
+        'executor': test_executor,
+        'score': 0,
+        'jitter': 0,
+        'cleanup': 0,
+        'pin': 0,
+        'host': test_agent.host,
+        'deadman': False,
+        'used': [],
+        'id': '456',
+        'relationships': []
+    }
+
+
+@pytest.fixture
+def finished_link(test_executor, test_agent, test_ability):
+    return {
+        'command': test_executor.command,
+        'paw': test_agent.paw,
+        'ability': test_ability,
+        'executor': test_executor,
+        'host': test_agent.host,
+        'deadman': False,
+        'used': [],
+        'id': '789',
+        'relationships': [],
+        'status': 0
+    }
+
+
+@pytest.fixture
+def setup_finished_operation(loop, expected_operation):
+    finished_operation = OperationSchema().load(expected_operation)
+    finished_operation.id = '000'
+    finished_operation.state = 'finished'
+    loop.run_until_complete(BaseService.get_service('data_svc').store(finished_operation))
+
+
+@pytest.fixture
+def setup_operations_api_test(loop, api_client, expected_operation, test_agent, test_ability,
+                              active_link, finished_link):
+    test_operation = OperationSchema().load(expected_operation)
+    test_operation.agents.append(test_agent)
+    test_operation.set_start_details()
+    test_operation.adversary.atomic_ordering.append(test_ability)
```

Comment:
is this necessary since it's supposed to be an ad-hoc adversary?

Reply:
That's a good point, I'll remove that